pkg/nfd-master/nfd-master.go

```diff
@@ -514,11 +636,14 @@ func (m *nfdMaster) crLabels(r *pb.SetLabelsRequest) map[string]string {
 			klog.Infof("executing NodeFeatureRule %q", spec.ObjectMeta.Name)
 		}
 		for _, rule := range spec.Spec.Rules {
-			ruleOut, err := rule.Execute(features)
+			featureMatched, anyMatched, ruleOut, err := rule.Execute(features)
 			if err != nil {
 				klog.Errorf("failed to process Rule %q: %v", rule.Name, err)
 				continue
 			}
+			if featureMatched || anyMatched {
+				crt = append(crt, rule.Taints...)
+			}
```

Comment:
We shouldn't need these `featureMatched` and `anyMatched` arguments at all. We should have all taints in `ruleOut`, right (i.e. just unconditional `crt = append(crt, ruleOut.Taints`)?

Reply:
That's right. I just realized that I misunderstood the matched/unmatched rule logic but thanks to your comment I understand it now. Please have a look once more.